adapters/sharethrough/utils.go

```diff
@@ -183,17 +193,23 @@ func (u Util) gdprApplies(request *openrtb.BidRequest) bool {
 	return gdprApplies != 0
 }
 
-func (u Util) gdprConsentString(request *openrtb.BidRequest) string {
-	var consentString string
-
-	if request.User != nil {
-		if jsonExtUser, err := request.User.Ext.MarshalJSON(); err == nil {
-			// empty string is the return value if error, so no need to handle
-			consentString, _ = jsonparser.GetString(jsonExtUser, "consent")
+func (u Util) parseUserExt(user *openrtb.User) userInfo {
```

Comment:
You could make this `func (u Util) parseUserExt(user *openrtb.User) (ui userInfo) {` and then use `ui.Consent` and `ui.TtdUid`instead of `unifiedId` and `gdprConsentString`. Then you just need to `return` rather than building a `userInfo` struct in the return statement.

Reply:
Good to know, thanks for the tip. I'm just gonna leave it this way for now as I don't want to cancel your approval and wait for another round of review 😆I'll likely address it with the next feature/PR 👍